packages/core/review-workflows/server/src/register.ts

```diff
@@ -56,33 +54,62 @@ function extendReviewWorkflowContentTypes({ strapi }: { strapi: Core.Strapi }) {
   }
 }
 
-// TODO: V5
-// function persistStagesJoinTables({ strapi }: { strapi: Core.LoadedStrapi }) {
-//   return async ({ contentTypes }: any) => {
-//     const getStageTableToPersist = (contentTypeUID: any) => {
-//       // Persist the stage join table
-//       const { attributes, tableName } = strapi.db.metadata.get(contentTypeUID) as any;
-//       const joinTableName = attributes[ENTITY_STAGE_ATTRIBUTE].joinTable.name;
-//       return {
-//         name: joinTableName,
-//         dependsOn: [{ name: tableName }],
-//       };
-//     };
-
-//     const joinTablesToPersist = pipe([
-//       getVisibleContentTypesUID,
-//       filter((uid: any) => hasStageAttribute(contentTypes[uid])),
-//       map(getStageTableToPersist),
-//     ])(contentTypes);
-
-// await removePersistedTablesWithSuffix('_strapi_stage_links');
-// await persistTables(joinTablesToPersist);
-// };
-// }
+/**
+ * Persist the stage & assignee attributes so they are not removed when downgrading to CE.
+ *
+ * TODO: V6 - Instead of persisting the join tables, always create the stage & assignee attributes, even in CE mode
+ *            It was decided in V4 & V5 to not expose them in CE (as they pollute the CTs) but it's not worth given the complexity this needs
+ */
+function persistRWOnDowngrade({ strapi }: { strapi: Core.Strapi }) {
+  const { removePersistedTablesWithSuffix, persistTables } = getAdminService('persist-tables');
+
+  return async ({ contentTypes }: any) => {
```

Comment:
We don't have any types for contentTypes?

Reply:
we do! will add it :)